core/remote_state_verifier.go

```diff
@@ -60,23 +62,25 @@ func NewVerifyManager(blockchain *BlockChain, peers verifyPeers, allowInsecure b
 	if block == nil {
 		return nil, ErrCurrentBlockNotFound
 	}
-	number := block.Number()
-	for i := maxForkHeight; i >= 0; i-- {
-		if new(big.Int).Sub(number, big.NewInt(int64(i))).Cmp(common.Big0) <= 0 {
-			continue
-		}
-		oldBlock := blockchain.GetBlockByNumber(number.Uint64() - uint64(i))
-		if oldBlock == nil {
-			return nil, fmt.Errorf("block is nil, number: %d", number)
-		}
-		blockHash := oldBlock.Hash()
-		_, err := blockchain.GenerateDiffLayer(blockHash)
-		if err != nil {
-			return nil, err
+
+	// rewind to last non verified block
+	number := new(big.Int).Sub(block.Number(), big.NewInt(int64(maxForkHeight)))
+	if number.Cmp(common.Big0) < 0 {
+		blockchain.SetHead(0)
+	} else {
+		numberU64 := number.Uint64()
+		blockchain.SetHead(numberU64)
+		block := blockchain.GetBlockByNumber(numberU64)
+		for i := 0; i < maxForkHeight; i++ {
```

Comment:
There may not enough parent blocks,  i < maxForkHeight && block.Number().Uint64 > 0 ?

Reply:
fixed